src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java

```diff
@@ -126,450 +80,275 @@ public ElasticsearchRestTemplate(RestHighLevelClient client, ElasticsearchConver
 
 	private void initialize(RestHighLevelClient client, ElasticsearchConverter elasticsearchConverter) {
 		Assert.notNull(client, "Client must not be null!");
-		this.client = client;
-		initialize(elasticsearchConverter);
+		initialize(elasticsearchConverter, //
+				new DefaultIndexOperations(client, elasticsearchConverter), //
+				new DefaultDocumentOperations(client, elasticsearchConverter), //
+				new DefaultSearchOperations(client, elasticsearchConverter));
 	}
+	// endregion
 
+	// region IndexOperations
+
+	/**
+	 * @deprecated use {@link #getIndexOperations()} and {@link IndexOperations#createIndex(String, Object)}.
+	 */
 	@Override
-	public boolean addAlias(AliasQuery query, IndexCoordinates index) {
-		IndicesAliasesRequest request = requestFactory.indicesAddAliasesRequest(query, index);
-		try {
-			return client.indices().updateAliases(request, RequestOptions.DEFAULT).isAcknowledged();
-		} catch (IOException e) {
-			throw new ElasticsearchException("failed to update aliases with request: " + request, e);
-		}
+	@Deprecated
+	public boolean createIndex(String indexName, Object settings) {
```

Comment:
Deprecations should happen on the interface. The implementation is typically not visible to users as we recommend using the `ElasticsearchOperations` interface directly.

Reply:
done